crates/bevy_ecs/src/system/system_registry.rs

```diff
@@ -500,6 +500,16 @@ impl World {
 
     /// Runs a cached system, registering it if necessary.
     ///
+    /// # Type Inference Note
+    /// If the system returns `()`, you may need to explicitly constrain the output
```

Comment:
I think inference works for system functions that return plain `()`, and we only get errors for systems that return `Result`.  

```suggestion
    /// If the system returns `Result`, you may need to explicitly constrain the output
```